cpp/src/arrow/compute/kernel.h

```diff
@@ -555,26 +555,51 @@ struct ARROW_EXPORT Kernel {
 /// employed this may not be possible.
 using ArrayKernelExec = Status (*)(KernelContext*, const ExecSpan&, ExecResult*);
 
+/// \brief The optional scalar kernel selective execution API for SCALAR kernel types.
+/// It's like ArrayKernelExec but with an additional SelectionVectorSpan argument. When a
+/// selection vector is specified in the batch, this API will be preferred, if provided,
+/// over ArrayKernelExec.
+using ArrayKernelSelectiveExec = Status (*)(KernelContext*, const ExecSpan&,
+                                            const SelectionVectorSpan&, ExecResult*);
```

Comment:
I don't have a suggestion yet, but if we plan to support bitmaps as well, it would probably be better to pass something here that can be either a selection vector or a bitmap mask. The alternative being yet another `KernelExec` -- `ArrayKernelMaskedExec`.

Reply:
Selection vectors are better than bitmask for very selective filters. Bitmasks are better when the filter is not very selective. Bitmaps are less important than selection vectors because if the filter is not selected computing on every value is not as bad.